presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/joins/JoinGraph.java

```diff
@@ -14,6 +14,7 @@
 package com.facebook.presto.sql.planner.optimizations.joins;
```

Comment:
You mean "Preserve leaf GroupReferences in EliminateCrossJoins rule" commit? I had this separate because

- without this commit code is still correct. It didn't lookup whole plan all the way down. It just looked up one level too far.
- test in this commit requires changes to rule testing framework (tests need to use lookup, we need to be able to match `GroupReference`-s and print them)
- i hoped reviewing will be easier

Anyway, if all gets accepted, let's squash. I renamed the commit to fixup.